source/extensions/filters/network/generic_proxy/interface/codec.h

```diff
@@ -22,12 +22,20 @@ class ServerCodec {
   virtual ~ServerCodec() = default;
 
   /**
-   * Set callbacks of server codec.
+   * Set callbacks of server codec. Called before onDownstreamConnected().
    * @param callbacks callbacks of server codec. This callback will have same or longer
    * lifetime as the server codec.
    */
   virtual void setCodecCallbacks(ServerCodecCallbacks& callbacks) PURE;
 
+  /**
+   * Called when the downstream connection is established.
+   *
+   * The connection obtained from ServerCodecCallbacks::connection() will be valid when this
+   * callback is invoked. It should not be relied upon to be valid until this point.
+   */
+  virtual void onDownstreamConnected() {}
```

Comment:
How about just name it `onConnected()` (simple) or `onNewConnection()` (keep same with the network filter)?